### ISSUE TITLE
useMutation expects incorrect mutateOptions as argument on mutate

### ISSUE THREAD

**Issue by basslagter** [tannerlinsley/react-query#661]
**Describe the bug**
Typescript complains about the argument I provide to my mutate function. This is since upgrade to the 2.4.4 version today

**To Reproduce**
```
 const [remove, { status }] = useMutation(removeSomething, {
    onSuccess: () => {},
    onError: () => {},
  });

```
When I call the mutate method like so:
`remove('someId'); `

I get **'Type string has no properties in common with type 'MutateOptions''**

The removeSomething function looks like:

```
export async function removeSomething(idToRemove: string): Promise<void> {
  // fetch
}
```

**Comment by Habbeunik** [comment]
Can you please specify the exact version, I just tested with 2.4.4 and I don't seem to have anything  error

**Comment by basslagter** [comment]
Update above. It's version 2.4.4 with typescript  2.9.5.
I use it with a NextJs app (9.4.4) if that might be of any influence.

**Comment by youngbob** [comment]
This is happening to me too. I think it is due to https://github.com/tannerlinsley/react-query/blob/b3e4dda9fd80ec6f0ade49034ae6eeafe2055942/types/index.d.ts#L446 - if strictNullChecks is off, then `undefined` extends everything. Patching locally to reverse this (`TVariables extends undefined`) fixes it for me, though I'm not sure if that's actually the correct fix or not.

**Comment by basslagter** [comment]
2.4.5 doesn't fix it either

**Comment by jelteliekens** [comment]
@youngbob What you're saying is correct. It is best to reverse it so that TVariables extends undefined and not the other way arround. I created a pull request for it.

**Comment by basslagter** [comment]
With 2.4.6 it's still not fixed for me. Same message

**Comment by lee-reinhardt** [comment]
I don't think the fix was included in 2.4.6:
https://github.com/tannerlinsley/react-query/blob/v2.4.6/types/index.d.ts#L447
